google-cloud-resourcemanager/src/main/java/com/google/cloud/resourcemanager/PolicyMarshaller.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.resourcemanager;
+
+import com.google.api.services.cloudresourcemanager.model.Binding;
+import com.google.cloud.Identity;
+import com.google.cloud.Policy;
+import com.google.cloud.Policy.Marshaller;
+import com.google.cloud.Role;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+final class PolicyMarshaller
+    extends Marshaller<com.google.api.services.cloudresourcemanager.model.Policy> {
+
+  static final PolicyMarshaller INSTANCE = new PolicyMarshaller();
+
+  private PolicyMarshaller() {}
+
+  private static class Builder extends com.google.cloud.Policy.Builder {
```

Comment:
I am not sure this need justify creating a subclass of `com.google.cloud.Policy.Builder` rather than setting the value directly on `Policy.builder()` (and in that case making that Builder final instead of some of its methods).